### PR TITLE
#132 Deaths Extract Processing Conversion

### LINKED ISSUE #132: Convert (IT) Deaths extract/lookup processing
@catherineholland1 is starting this.

### PR DESCRIPTION
**Pull request by catherineholland1**
Closes #132 Deaths Extract Conversion

Lookup-04 conversion

### CODE REVIEW

File: DEATHS_extract_processing.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                     data$death_date_NRS[i],
+                     data$death_date_CHI[i])
+}
+
+
+
+## Save file ##
+# //stats/hscdiip/SLF_extracts/Deaths/ #
+# get_slf_deaths_path()
+saveRDS(data, get_slf_deaths_path())
```
Comment (Jennit07) [inline DEATHS_extract_processing.R:40]: Remember to remove unrelated comments when you are finished. Do we need the save RDS and write_rds? or are they different?

**Review by Jennit07 — CHANGES_REQUESTED** [review]
Looks good - a few changes suggested :)

File: DEATHS_extract_processing.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
+
+
+
```
[the remark this answers is not in the document]
Comment (catherineholland1) [reply DEATHS_extract_processing.R:3]: comments added

File: DEATHS_extract_processing.R [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Read data in ##
+data <- readr::read_csv(get_it_deaths_path())
+
+
+# rename
+colnames(data) = c("chi",
+                   "death_date_NRS",
+                   "death_date_CHI")
```
[the remark this answers is not in the document]
Comment (catherineholland1) [reply DEATHS_extract_processing.R:11]: data reading in and renaming changed to fit with formatting

File: DEATHS_extract_processing.R [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+
+## NRS death date if avaliable ##
+for (i in 1:nrow(data)) {
+  data$death_date[i] <- ifelse(is.na(data$death_date_NRS[i]) == FALSE,
+                     data$death_date_NRS[i],
+                     data$death_date_CHI[i])
+}
+
```
[the remark this answers is not in the document]
Comment (catherineholland1) [reply DEATHS_extract_processing.R:34]: combined lines of code within %>% to use NRS death date in new variable

File: DEATHS_extract_processing.R [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+# //stats/hscdiip/SLF_extracts/Deaths/ #
+# get_slf_deaths_path()
+saveRDS(data, get_slf_deaths_path())
+
+# haven
+haven::write_sav(LTC, get_slf_deaths_path(), compress = TRUE
+)
+
+#readr
+readr::write_rds(LTC, get_slf_deaths_path(), compress = "gz")
```
[the remark this answers is not in the document]
Comment (catherineholland1) [reply DEATHS_extract_processing.R:47]: changed typo

**Review by catherineholland1 — COMMENTED** [review]
edits complete

@Jennit07 review again

File: DEATHS_extract_processing.R [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+## packages ##
+
+library(dplyr)
+library(stringr)
+library(readr)
+
+
+
+## Read data in ##
+data <- readr::read_csv(get_it_deaths_path())
```
Comment (Jennit07) [inline DEATHS_extract_processing.R]: I've removed this line as you have it underneath and also just removed spaces within the script. But looks good to me :)

**Review by Jennit07 — APPROVED** [review]

File: DEATHS_extract_processing.R [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    chi = "PATIENT_UPI_[C]",
+    death_date_NRS = "PATIENT_DoD_DATE_(NRS)",
+    death_date_CHI = "PATIENT_DoD_DATE_(CHI)"
+  )
+
+## one record per chi ##
+deaths_file <- deaths_file %>%
+  dplyr::distinct(chi, .keep_all = TRUE) %>%
+  # Use the NRS deathdate unless it isn't there
+  mutate(death_date = if_else(is.na(death_date_NRS), death_date_CHI, death_date_NRS))
```
Comment (Moohan) [inline DEATHS_extract_processing.R:46]: distinct will just keep the first row, so we should sort first to make sure we are not dropping a date if there is one
```suggestion
deaths_file <- deaths_file %>%
  dplyr::arrange(desc(death_date_NRS), desc(death_date_CHI)) %>%
  dplyr::distinct(chi, .keep_all = TRUE) %>%
  # Use the NRS deathdate unless it isn't there
  mutate(death_date = if_else(is.na(death_date_NRS), death_date_CHI, death_date_NRS))
```

Comment (Moohan) [inline DEATHS_extract_processing.R:46]: `dplyr::coalesce()` does exactly what we want for the last line i.e. pick the first non-missing date.
```suggestion
deaths_file <- deaths_file %>%
  dplyr::distinct(chi, .keep_all = TRUE) %>%
  # Use the NRS deathdate unless it isn't there
  mutate(death_date = dplyr::coalesce(death_date_NRS, death_date_CHI)
```

Comment (catherineholland1) [reply DEATHS_extract_processing.R:46]: included to arrange to make sure the date isn't lost

**Comment by catherineholland1** [comment]
ready to be merged? @Jennit07 @jam

File: DEATHS_extract_processing.R [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    death_date_nrs = "PATIENT_DoD_DATE_(NRS)",
+    death_date_chi = "PATIENT_DoD_DATE_(CHI)"
+  )
+
+## one record per chi ##
+deaths_file <- deaths_file %>%
+  dplyr::arrange(desc(death_date_NRS), desc(death_date_CHI)) %>%
+  dplyr::distinct(chi, .keep_all = TRUE) %>%
+  # Use the NRS deathdate unless it isn't there
+  mutate(death_date = dplyr::coalesce(death_date_NRS, death_date_CHI))
```
Comment (Moohan) [inline DEATHS_extract_processing.R:47]:
```suggestion
deaths_file <- deaths_file %>%
  rename(
    chi = "PATIENT_UPI_[C]",
    death_date_nrs = "PATIENT_DoD_DATE_(NRS)",
    death_date_chi = "PATIENT_DoD_DATE_(CHI)"
  )

## one record per chi ##
deaths_file <- deaths_file %>%
  dplyr::arrange(desc(death_date_nrs), desc(death_date_chi)) %>%
  dplyr::distinct(chi, .keep_all = TRUE) %>%
  # Use the NRS deathdate unless it isn't there
  mutate(death_date = dplyr::coalesce(death_date_nrs , death_date_chi))
```

**Review by Moohan — CHANGES_REQUESTED** [review]

**Review by Moohan — APPROVED** [review]

**Pull request merged**